### PR TITLE
Fix `app import` failure.

### LINKED ISSUE #9691: [mesheryctl] Failure to import Kubernetes Manifest with `app import`
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior
While using a valid kuberenetes manifest (`constraint.yaml`), the `mesheryctl app import` command fails to import the manifest.

```
mesheryctl app import -f constraint.yaml -s manifest
Error: Invalid application source type due to wrong type/passing
```

### Environment

- Host OS: Mac 

```
      	VERSION	GITSHA
Client	v0.7.1 	ef8db2827c
Server	v0.7.3 	Not Set
```

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by shanukun**
Use `/api/pattern/types` endpoint to request source types instead of `/api/application/types`.

**Notes for Reviewers**

This PR fixes #9691

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/meshery_application.go [hunk 1]
```diff
@@ -15,17 +15,17 @@ type ApplicationTypeResponse struct {
 	SupportedExtensions []string        `json:"supported_extensions"`
 }
 
-func GetApplicationTypes() (r []DesignTypeResponse) {
-	r = append(r, DesignTypeResponse{
-		Type:                HelmChart,
+func GetApplicationTypes() (r []ApplicationTypeResponse) {
+	r = append(r, ApplicationTypeResponse{
+		Type:                ApplicationType(HelmChart),
```
Comment (shanukun) [inline server/models/meshery_application.go:20]: @theBeginner86, one of the possible solution is to make it return []ApplicationTypeResponse and explicity converting the DesignType to ApplicationType.

**Review by theBeginner86 — CHANGES_REQUESTED** [review]
Thanks for this.

As I mentioned earlier on the issue, please consider using the `DesignTypeResponses` construct in `mesheryctl` and don't revert back to `ApplicationTypeResponses`. Otherwise it would only lead to confusion as backend no more explicitly calls out Applications and all UI clients requests /patterns handlers due to recent migration.

**Comment by shanukun** [comment]
> all UI clients requests /patterns handlers due to recent migration.

@theBeginner86 In that case, shouldn't we drop `/api/application/types` endpoint and use `/api/pattern/types` for requesting source types along with a more pertinent struct to unmarshal the json respone.

**Comment by theBeginner86** [comment]
> > all UI clients requests /patterns handlers due to recent migration.
> 
> @theBeginner86 In that case, shouldn't we just drop `/api/application/types` endpoint and use `/api/pattern/types` for requesting source types along with a more pertinent struct to unmarshal the json respone.

yes, we should. Would you like to take a crack at it and ensure that mesheryctl uses that API?

**Comment by theBeginner86** [comment]
Added a couple of updates to import 👍

**Review by theBeginner86 — APPROVED** [review]
Thanks @shanukun. LGTM

Screenshot 2024-01-01 at 12 45 12 PM

> _(neglect the extraneous debug log. I had added for testing. Its not part of the actual output)_

**Comment by theBeginner86** [comment]
Lets merge this....

I am creating a new PR to update rest of the app commands and test cases 👍

**Pull request merged**